server/build/index.js

```diff
@@ -19,26 +21,34 @@ export default async function build (dir, conf = null) {
     throw err
   }
 
-  const compiler = await webpack(dir, { buildId, buildDir, conf })
-
   try {
-    const stats = await runCompiler(compiler)
-    await writeBuildStats(buildDir, stats)
-    await writeBuildId(buildDir, buildId)
+    const configs = await Promise.all([
+      getBaseWebpackConfig(dir, { buildId, isServer: false, config }),
+      getBaseWebpackConfig(dir, { buildId, isServer: true, config })
+    ])
+
+    await runCompiler(configs)
+
+    // await fs.writeFile(join(dir, '.next', 'server-stats.json'), JSON.stringify(serverStats), 'utf8')
+    // await fs.writeFile(join(dir, '.next', 'client-stats.json'), JSON.stringify(stats), 'utf8')
+
+    await writeBuildStats(dir)
+    await writeBuildId(dir, buildId)
   } catch (err) {
-    console.error(`> Failed to build on ${buildDir}`)
+    console.error(`> Failed to build`)
     throw err
   }
 
-  await replaceCurrentBuild(dir, buildDir)
+  // await replaceCurrentBuild(dir, buildDir)
 
   // no need to wait
-  del(buildDir, { force: true })
+  // del(buildDir, { force: true })
```

Comment:
Don't we need to delete the current directory before running next build.
Is webpack doing that already?